From 2c1060ddb13da2a4f023c082c1534f8e3222156d Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Tue, 12 Nov 2024 10:02:04 -0700
Subject: [PATCH 06/16] SUBMITTED: LoongArch: KVM: Ensure ret is always
 initialized in kvm_eiointc_{read,write}()

Clang warns (or errors with CONFIG_WERROR=y):

  arch/loongarch/kvm/intc/eiointc.c:323:2: error: variable 'ret' is used uninitialized whenever switch default is taken [-Werror,-Wsometimes-uninitialized]
    323 |         default:
        |         ^~~~~~~
  arch/loongarch/kvm/intc/eiointc.c:697:2: error: variable 'ret' is used uninitialized whenever switch default is taken [-Werror,-Wsometimes-uninitialized]
    697 |         default:
        |         ^~~~~~~

Set ret to -EINVAL in the default case to resolve the warning, as len
was not a valid value for the functions to handle.

Fixes: e24e9e0c1da4 ("LoongArch: KVM: Add EIOINTC read and write functions")
Link: https://lore.kernel.org/r/20241112-loongarch-kvm-eiointc-fix-sometimes-uninitialized-v1-1-7d881f728d67@kernel.org
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 arch/loongarch/kvm/intc/eiointc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/loongarch/kvm/intc/eiointc.c b/arch/loongarch/kvm/intc/eiointc.c
index 0084839f4150..6af3ecbe29ca 100644
--- a/arch/loongarch/kvm/intc/eiointc.c
+++ b/arch/loongarch/kvm/intc/eiointc.c
@@ -323,6 +323,7 @@ static int kvm_eiointc_read(struct kvm_vcpu *vcpu,
 	default:
 		WARN_ONCE(1, "%s: Abnormal address access: addr 0x%llx, size %d\n",
 						__func__, addr, len);
+		ret = -EINVAL;
 	}
 	spin_unlock_irqrestore(&eiointc->lock, flags);
 
@@ -697,6 +698,7 @@ static int kvm_eiointc_write(struct kvm_vcpu *vcpu,
 	default:
 		WARN_ONCE(1, "%s: Abnormal address access: addr 0x%llx, size %d\n",
 						__func__, addr, len);
+		ret = -EINVAL;
 	}
 	spin_unlock_irqrestore(&eiointc->lock, flags);
 
-- 
2.47.0

